src/Build/Evaluation/LazyItemEvaluator.cs

```diff
@@ -291,6 +291,9 @@ public ImmutableList<ItemData>.Builder GetItemData(ImmutableHashSet<string> glob
                 }
             }
 
+            /// <summary>
+            /// Applies uncached item operations (include, remove, update) in order.
```

Comment:
```suggestion
            /// Applies uncached item operations (include, remove, update) in order. Since Remove effectively overwrites Include or Update,
            /// Remove operations are preprocessed (adding to globsToIgnore) to create a longer list of globs we don't need to process
            /// properly because we know they will be removed. Update operations are batched as much as possible, meaning rather
            /// than being applied immediately, they are combined into a dictionary of UpdateOperations that need to be applied. This
            /// is to optimize the case in which as series of UpdateOperations, each of which affects a single ItemSpec, are applied to all
            /// items in the list, leading to a quadratic-time operation.
```